modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -2465,6 +2482,8 @@ private JavaObjectSerializer h2Serializer() {
 
     /** {@inheritDoc} */
     @Override public void unregisterCache(GridCacheContextInfo cacheInfo, boolean rmvIdx) {
+        cancelIndexRebuildFuture(idxRebuildFuts.remove(cacheInfo.cacheId()));
```

Comment:
now we can paused here, is it ok ? we need to complete future in this case, isn`t it ?